detector.go

```diff
@@ -119,6 +119,33 @@ func (c *Client) GetDetector(ctx context.Context, id string) (*detector.Detector
 	return finalDetector, err
 }
 
+// GetDetectors gets all detectors.
```

Comment:
it gets detectors, with a given offset and limit. Can we set a max limit?

Reply:
Per API [documentation](https://dev.splunk.com/observability/reference/api/detectors/latest), the max limit is 10000. I didn't add validation for it, but as previously mentioned in my other comment, I can add it if it's required/recommended.

Reply:
ok